src/main/kotlin/net/maizegenetics/phgv2/pathing/AlignmentUtils.kt

```diff
@@ -0,0 +1,292 @@
+package net.maizegenetics.phgv2.pathing
+
+import it.unimi.dsi.fastutil.longs.Long2LongOpenHashMap
+import kotlinx.coroutines.channels.ReceiveChannel
+import kotlinx.coroutines.channels.SendChannel
+import net.maizegenetics.phgv2.api.HaplotypeGraph
+import net.maizegenetics.phgv2.api.ReferenceRange
+import net.maizegenetics.phgv2.utils.getBufferedWriter
+import net.maizegenetics.phgv2.utils.getBufferedReader
+import org.apache.logging.log4j.LogManager
+import java.io.*
+import java.util.*
+import kotlin.math.ceil
+import kotlin.math.min
+
+/**
+ * data class to hold the KmerMap information.  From the PHGv1 source code:
+ */
+//data class KmerMapData(val haplotypeListId: Int, val rangeToBitSetMap: Map<Int, BitSet>, val kmerHashToLongMap: Long2LongOpenHashMap)
+data class KmerMapData(val rangeToBitSetMap: Map<ReferenceRange, BitSet>, val kmerHashToLongMap: Long2LongOpenHashMap)
+
+data class KeyFileData(val sampleName: String, val file1: String, val file2: String = "")
+
+private val myLogger = LogManager.getLogger("net.maizegenetics.phgv2.utils.AlignmentUtils")
+
+
+fun alignReadsToHaplotypes(hvcfDir: String, kmerIndexFile:String, keyFileRecords:List<KeyFileData>, paired:Boolean, outputDir:String) {
+    //loop through all files in hvcfDir and create a list of hvcf files
+    val hvcfFiles = File(hvcfDir).walkTopDown().filter { it.isFile }.filter { it.extension == "h.vcf" }.map { "${it.path}/${it.name}" }.toList()
```

Comment:
Just curious - will we require extension to always be h.vcf?  Could they be gzipped and be h.vcf.gz, or without the first period, e.g. hvcf or hvcf.gz?